crates/turbo-tasks-fetch/src/lib.rs

```diff
@@ -36,11 +36,34 @@ impl HttpResponseBody {
     }
 }
 
+#[turbo_tasks::value(shared)]
+#[derive(Debug)]
+pub enum ProxyConfig {
+    Http(String),
+    Https(String),
+}
+
+#[turbo_tasks::value(transparent)]
+pub struct OptionProxyConfig(Option<ProxyConfig>);
+
 #[turbo_tasks::function]
-pub async fn fetch(url: Vc<String>, user_agent: Vc<Option<String>>) -> Result<Vc<FetchResult>> {
+pub async fn fetch(
+    url: Vc<String>,
+    user_agent: Vc<Option<String>>,
+    proxy_option: Vc<OptionProxyConfig>,
```

Comment:
does it not work without the newtype?

Reply:
I got some build error message like cannot be awaited without intofuture trait impl, not sure what's the difference.

Reply:
```
error[E0277]: the trait bound `Vc<std::option::Option<ProxyConfig>>: IntoFuture` is not satisfied
  --> crates/turbo-tasks-fetch/src/lib.rs:57:39
   |
57 |     let proxy_option = &*proxy_option.await?;
   |                                      -^^^^^
   |                                      ||
   |                                      |the trait `IntoFuture` is not implemented for `Vc<std::option::Option<ProxyConfig>>`
   |                                      help: remove the `.await`
   |
   = help: the following other types implement trait `IntoFuture`:
             Vc<T>
             &Vc<T>
```

Reply:
I think you still have to wrap the inner type in a vc
`Vc<Option<Vc<ProxyConfig>>`

But I guess that makes it a bit more annoying to use

Reply:
Looks like so. Probably ok to have a new type?